flower/events.py

```diff
@@ -6,28 +6,26 @@
 
 from functools import partial
 
-import celery
-
 from tornado.ioloop import IOLoop
 from tornado.ioloop import PeriodicCallback
-from tornado.concurrent import run_on_executor
 
 from celery.events import EventReceiver
 from celery.events.state import State
 
 from . import api
 
 from collections import Counter
-from concurrent.futures import ThreadPoolExecutor
 
-from prometheus_client import Counter as PrometheusCounter, Histogram
+from prometheus_client import Counter as PrometheusCounter, Histogram, Gauge
 
 logger = logging.getLogger(__name__)
 
 
 class PrometheusMetrics(object):
     events = PrometheusCounter('flower_events_total', "Number of events", ['worker', 'type', 'task'])
     runtime = Histogram('flower_task_runtime_seconds', "Task runtime", ['worker', 'task'])
+    queuing_time = Gauge('flower_task_queuing_time_seconds', "Task queueing time", ['worker', 'task'])
```

Comment:
I felt that for those new metrics I prefer current, most accurate measurement rather than a histogram.

Reply:
`task-received - task-started` is the time task spends in a worker waiting to run. The time task spends in broker will not be included, so `queuing_time` name is a bit misleading.

Reply:
Yeah indeed. The name is not ideal but I guess it is a good approximation. In general I expect my broker to act nearly instantly and it has its own metrics so I feel covered there. My inspiration was code in flower 0.9.4:

`queue_time += task.started - task.received`, but I am all for 'explicit is better than implicit', so what about: `flower_task_queuing_time_at_worker_seconds` or `flower_task_waiting_time_at_worker` ? 

Personally I like the word `queuing` in there.

I also noticed that we were avoiding calculating this `queue time` if task had an `eta` set on it as it makes no sense (it will be waiting at the worker then). I could for those cases calculate queuing time as: `task.started - task.eta` showing what was the delay in actually starting the task past the initial eta value or skip completely as it was done in Flower 0.9.4 where we have:

```
    if task.eta is not None:
        continue
```

What do you think is better?

Reply:
I have changed to: `flower_task_queuing_time_at_worker_seconds`.

Reply:
The question still stands about the `eta`. I never used it (prefer cron to schedule stuff) but if you think it's best to skip computing queuing time for eta tasks please let me know.

Reply:
I think it makes sense to filter out the tasks with eta

Reply:
I will follow up with some handling of the ETA then similar to what Flower did in 0.9.4.